python/lsst/sims/utils/FocalPlaneUtils.py

```diff
@@ -0,0 +1,237 @@
+    transforations (in years)
+
+    @param [out] a 2-D numpy array in which the first row is RA and the second
+    row is Dec (both in degrees; both in the International Celestial Reference System)
+    """
+
+    output = _raDecFromPupilCoords(xPupil, yPupil, obs_metadata=obs_metadata,
+                                               epoch=epoch)
+
+    return np.degrees(output)
+
+
+def _raDecFromPupilCoords(xPupil, yPupil, obs_metadata=None, epoch=None):
+    """
+    @param [in] xPupil -- pupil coordinates in radians
+
+    @param [in] yPupil -- pupil coordinates in radians
+
+    @param [in] obs_metadata -- an instantiation of ObservationMetaData characterizing
+    the state of the telescope
+
+    @param [in] epoch -- julian epoch of the mean equinox used for the coordinate
+    transforations (in years)
+
+    @param [out] a 2-D numpy array in which the first row is RA and the second
+    row is Dec (both in radians; both in the International Celestial Reference System)
+    """
+
+    if obs_metadata is None:
+        raise RuntimeError("Cannot call raDecFromPupilCoords without obs_metadata")
+
+    if epoch is None:
+        raise RuntimeError("Cannot call raDecFromPupilCoords; epoch is None")
+
+    if obs_metadata.rotSkyPos is None:
+        raise RuntimeError("Cannot call raDecFromPupilCoords without rotSkyPos " + \
+                           "in obs_metadata")
+
+    if obs_metadata.pointingRA is None or obs_metadata.pointingDec is None:
+        raise RuntimeError("Cannot call raDecFromPupilCoords "+ \
+                          "without pointingRA, pointingDec in obs_metadata")
+
+    if obs_metadata.mjd is None:
+        raise RuntimeError("Cannot calculate x_pupil, y_pupil without mjd " + \
+                           "in obs_metadata")
+
+    if len(xPupil)!=len(yPupil):
+        raise RuntimeError("You passed %d RAs but %d Decs into raDecFromPupilCoords" % \
+                           (len(raObj), len(decObj)))
+
+    ra_pointing_temp, dec_pointing_temp = _observedFromICRS(np.array([obs_metadata._pointingRA]),
+                                                            np.array([obs_metadata._pointingDec]),
+                                                            obs_metadata=obs_metadata,
+                                                            epoch=2000.0, includeRefraction=True)
+
+    ra_pointing = ra_pointing_temp[0]
+    dec_pointing = dec_pointing_temp[0]
+
+    #This is the same as theta in pupilCoordsFromRaDec, except without the minus sign.
+    #This is because we will be reversing the rotation performed in that other method.
+    theta = -1.0*obs_metadata._rotSkyPos
+
+    x_g = xPupil*np.cos(theta) - yPupil*np.sin(theta)
+    y_g = xPupil*np.sin(theta) + yPupil*np.cos(theta)
+
+    x_g *= -1.0
+
+    # x_g and y_g are now the x and y coordinates
+    # can now use the PALPY method palDtp2s to convert to RA, Dec.
+
+    raObs, decObs = palpy.dtp2sVector(x_g, y_g, ra_pointing, dec_pointing)
```

Comment:
Should there be a backup loop here like in the reverse transformation in case one of the array ra/dec values doesn't work?